src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Binding.cs

```diff
@@ -310,6 +319,15 @@ internal partial class CommonReferenceManager<TCompilation, TAssemblySymbol>
                 // Rebind assembly references that were initially missing. All bindings established above
                 // are against explicitly specified references.
 
+                // We only need to resolve against implicitly resolved assemblies,
+                // since we already resolved against explicitly specified ones.
+                var implicitAssembliesMap = new MultiDictionary<string, (AssemblyData DefinitionData, int DefinitionIndex)>(AssemblyIdentityComparer.SimpleNameComparer);
```

Comment:
>new MultiDictionary [](http://example.com/codeflow?start=44&length=19)

Can we also pass `capacity: implicitAssemblies.Count`? Similarly for `explicitAssembliesMap` above.